themes/default/content/blog/iac-recommended-practices-structuring-pulumi-projects/index.md

```diff
@@ -0,0 +1,104 @@
+
+Now that you have an idea of some of the factors that affect how users structure their Pulumi projects, what are the advantages and disadvantages of each approach?
+
+For single projects, the key advantage is _simplicity._ There's no denying that a single project is simpler and less complex than working with multiple projects. This advantage, though, is also the key disadvantage. A single project does not afford the flexibility to let different teams manage different aspects of the infrastructure, for example, nor does it offer the ability to split resources that don't have a shared lifecycle. And, as mentioned earlier, blast radius is a concern.
+
+For multiple projects, the key advantage is _flexibility._ It's true that the added flexibility comes at the cost of some added complexity (a future post will explore one aspect of that added complexity in the form of StackReferences). However, you gain the ability to separate infrastructure resources according to team ownership/responsibility, relationship to other resources (i.e., grouping resources in a "shared infrastructure layer"), or any of the other dimensions mentioned above.
+
+In some ways, the decision points for single vs. multiple projects are analogous to the decision points for monolith vs. microservice: independent scaling and flexibility versus simplicity.
+
+_In general, most customers end up using multiple Pulumi projects per use case._ As you've seen, though, there are a number of factors that influence that decision and will affect how many projects end up in use.
+
+## Moving forward with Zephyr
+
+Zephyr had a number of factors in play, not the least of which was the organizational change resulting from their growth and success. In the end, Zephyr decided to restructure their use of Pulumi for the online retail store into three separate projects:
+
+* An infrastructure project: This project handled the AWS VPC and related infrastructure resources.
+* A platform project: This project managed the Kubernetes cluster(s).
+* An application project: This project handled the actual online retail store itself.
+
+{{% notes %}}
+You can use a combination of code refactoring and `pulumi import` to assist in the process of restructuring Pulumi projects, but it some cases it may be easier to create new infrastructure and migrate over to it.
+{{% /notes %}}
+
+In [the first Zephyr post](/blog/iac-recommended-practices-code-organization-and-stacks/), you read about some of the options for where to store Pulumi code relative to the applications it supports. At that time, Zephyr had decided to go with what's called a monorepo approach, meaning that the Pulumi code was stored in the same repository as the application code. In conjunction with the project reorganization, Zephyr also decided to change how their code was organized:
+
+* Code for the infrastructure project moved to a separate "zephyr-infra" repository
+* A new "zephyr-k8s" repository housed the code for the platform project
```

Comment:
What would you think about calling these "network" and "platform"? You could imagine, for example, something other than Kubernetes acting as the deployment later but still residing in that same project, and the infra project looks like it's all about defining the networking. Thoughts?

Reply:
At this point renaming things will introduce additional delays that I'd rather avoid if possible. We can always show Zephyr using `pulumi stack rename` later and specifically call out to readers that this is how you rename projects and stacks to align with changes over time.